### PR TITLE
Feature/multicast delegate

### PR DESCRIPTION
**Pull request by maciejburda**
added multicast delegate for match delivery
tests improvements
updated readme
cleaned example (now I’m going to bring it back to correct API)

### CODE REVIEW

File: AlpsSDK/AlpsManager/AlpsManager+Convenience.swift [hunk 1]
```diff
@@ -10,41 +10,29 @@ import Foundation
...
                                         platform: device?.platform ?? uiDevice.systemName,
                                         deviceToken: device?.deviceToken ?? "",
                                         location: device?.location ?? location)
         mobileDevices.create(item: mobileDevice) { (result) in
-            if case let .success(createdDevice) = result {
-                completion?(createdDevice)
-            } else {
-                completion?(nil)
-            }
+            completion?(result)
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager+Convenience.swift:21]: I also think that giving back the result is better than just the object. The developer can know the error and something went wrong.

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 2]
```diff
@@ -10,7 +10,15 @@ import Foundation
 import CoreLocation
 import Alps
 
+public typealias OnMatchClojure = (_ matches: [Match], _ device: Device) -> Void
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift]: you mean closure no ?

File: README.md [hunk 3]
```diff
@@ -4,16 +4,16 @@
 
 ## Versioning
 
-Only compatible with Swift 3.
+SDK is written using Swift 4.
 
-Alps SDK requires iOS 7+, if you want to use iBeacons.
+Alps SDK requires iOS 9+.
```
Comment (wenjdu) [inline README.md]: Not sure but we did agree on 10+ ?

**Review by wenjdu — CHANGES_REQUESTED** [review]

File: AlpsSDK/AlpsManager/AlpsManager.swift (hunk 2 above)
Comment (maciejburda) [reply AlpsSDK/AlpsManager/AlpsManager.swift]: yeah :D good catch.

Mistake made because of https://clojure.org/

File: README.md (hunk 3 above)
Comment (maciejburda) [reply README.md]: I think you are right.

**Comment by maciejburda** [comment]
PR fixes added.

**Review by wenjdu — APPROVED** [review]

**Pull request merged**